src/translate.py

```diff
@@ -3350,6 +3359,10 @@ def handle_rlwinm(
     upper_bits: Optional[Expression]
     if left_mask == 0:
         upper_bits = None
+    elif isinstance(source, Literal) and (source.value << left_shift) & left_mask == (
+        source.value << left_shift
+    ):
+        upper_bits = Literal(source.value << left_shift)
```

Comment:
if `isinstance(source, Literal)` we should be able to compute the full result as `((source << shift) & mask) | ((source >> (32 - shift)) & mask)` and immediately return, rather than doing the upper_bits/lower_bits thing